### PR TITLE
Add builder macros

### PR DESCRIPTION
**Pull request by matthiasbeyer**
So this is my first attempt on #4.

The first commit compiles and tests successfully on my machine (using stable rust/cargo). The second fails, though I cannot see why.

Maybe some help from others, maybe @PlasmaPower or @Emerentius - one of you was pretty good with macros... and I guess I'm not that good :blush: :relaxed:

### CODE REVIEW

File: src/macros.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#[macro_export]
+macro_rules! construct_filter {
+
+    ( $a:ident $( and $e:ident )+ ) => {{
+        use $crate::ops::and::And;
+        And::new($a, construct_filter!( $( $e and )* ))
```
Comment (PlasmaPower) [inline src/macros.rs]: This shouldn't be a recursive object, it'll consume a ton of memory. I'd create a `MultiAnd<Vector<Filter>>` or similar.

File: src/macros.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }};
+    ( $a:ident and ) => {{ $a }};
+
+
+}
```
Comment (PlasmaPower) [inline src/macros.rs]: I'd also add an implementation `( $a: ident ) => {{ $a }};`.

Comment (PlasmaPower) [inline src/macros.rs]: And other operators of course.

Comment (matthiasbeyer) [inline src/macros.rs]: Yeah, the other operators are trivial as soon as I get one implemented, IMO. But I'm not sure how to do this properly... so, small steps here! :smile:

File: src/macros.rs (hunk 1 above)
Comment (matthiasbeyer) [inline src/macros.rs]: How'd I use the macro arguments to construct a `Vec` then?

Comment (PlasmaPower) [inline src/macros.rs]: `vec![$( $e, )*]`

Comment (matthiasbeyer) [inline src/macros.rs]: well, we have the problem that if we want to implement with a `Vec`, all arguments must be of the same type, right? That'd make it a lot less flexible.

Comment (PlasmaPower) [inline src/macros.rs]: `Vec<Box<Filter>>` or similar would still consume a lot less memory I think. You should probably keep this current implementation for 2 args though.

**Pull request closed without merging**